C++: Promote insecure removal of memset query (#5256)
github/codeql: cpp/change-notes/2021-02-24-memset-may-be-deleted.md

lgtm,codescanning

> * A new query (`cpp/memset-may-be-deleted`) is run and displayed on LGTM. The query finds calls to memset that may be removed by compiler. This query was originally submitted as an experimental query by @ihsinme in https://github.com/github/codeql/pull/4953.

Reviewer: Should a change note for `lgtm,codescanning` reference LGTM (and not codescanning) in the text?

Author: Yeah, maybe that's a bit weird. How about just writing "A new query (`cpp/memset-may-be-deleted`) has been added."?

Reviewer: Or "A new query (`cpp/memset-may-be-deleted`) has been added.  This query is run and displayed on LGTM [and run in the codescanning suite?]."

@jbj do you have an opinion on this?  This probably won't be the last time it comes up.

Reviewer 2: ```suggestion
* A new query (`cpp/memset-may-be-deleted`) is added to the default query suite. The query finds calls to `memset` that may be removed by compiler. This query was originally [submitted as an experimental query by @ihsinme](https://github.com/github/codeql/pull/4953).
```

I've suggested an edit to side-step the question for now, and then we can have the difficult discussion when we encounter a real example of a query that works differently on Code Scanning and LGTM.

> * A new query (`cpp/memset-may-be-deleted`) is added to the default query suite. The query finds calls to `memset` that may be removed by compiler. This query was originally [submitted as an experimental query by @ihsinme](https://github.com/github/codeql/pull/4953).

Reviewer 3: ```suggestion
* A new query (`cpp/memset-may-be-deleted`) is added to the default query suite. The query finds calls to `memset` that may be removed by the compiler. This behavior can make information-leak vulnerabilities easier to exploit. This query was originally [submitted as an experimental query by @ihsinme](https://github.com/github/codeql/pull/4953).
```

It might be helpful to describe the vulnerability here 😊
